Change line

nels3 opened the pull request:
Look at test node. I have implemented the demo version of code. I need to know if this seems ok. 
Please do: 
A) branch change_line 
B) roscore  
rosrun selfie_control demo_position (one command window) 
rosrun selfie_control change_lane _lane_width:=1 _error_margin:=0.1  (another command window, width positive, error_margin positive) 
rosrun selfie_control demo_change_lane _lane:=l (or _lane:=r, another command window, should run commands with l or r argument) 

 If this is what was needed we could talk about what should be pushed  to develop branch (I don't know about demo nodes and I will delete  ROS_INFO also inside it couse this is only for demo).

Goldob commented on src/selfie_control/src/ChangeLane.cpp:
```diff
@@ -0,0 +1,53 @@
+//output taget_positions, left_turn_indicator, right_turn_indicator
+//left +, right -
+bool ChangeLane::process_target_position(void){
+    if (be_on_left_lane == true){
```
Consider instead: `if (be_on_left_lane)`. This way it is a bit closer to natural language.

nels3 commented:
You should not analyze code now. You have told me what to do on tuesday and now I am trying to developer the code. So do not look at the latest commit. I do not know what will work, that's why there will be commented out code. 

Please wait with review till I will make changes connected with your comments from tuesday meeting.

Goldob commented:
I see. What confused me was your review request. I guess you issued the request before we talked. My bad :smile:

nels3 commented:
Exactly, when I requested there was working code that made sense. Now code doesn't work and I have changed a lot. Commit was oy for backup. When I will have code, I would let you know :)

nels3 commented:
Implemented new way of ChangeLaneAction. Please add your review.
have: float your_lane_width and float your_error_margin
Do:
roscore
rosrun selfie_control demo_position _lane_width:=your_lane_width
rosrun selfie_control change_lane _lane_width:=your_lane_width _error_margin:=your_error_margin
rosrun selfie_control demo_change_lane _lane:=r (or _lane:=l)

In demo_change_lane widow you will have information about status.
When Succeed - we finished maneuver
Answer 0 - we didn't change lane during maneuver
Answer 1 - we changed lane during maneuver
